source/as-stream.ts

```diff
@@ -121,10 +122,11 @@ export default function asStream(options: MergedOptions) {
 
 	proxy.on('pipe', source => {
 		if (source instanceof IncomingMessage) {
+			// eslint-disable-next-line @typescript-eslint/no-object-literal-type-assertion
 			options.headers = {
 				...source.headers,
 				...options.headers
-			};
+			} as Record<string, string | string[]>;
```

Comment:
For reference, the Node.js interface for headers is as follows:

```ts
// outgoing headers allows numbers (as they are converted internally to strings)
    interface OutgoingHttpHeaders {
        [header: string]: number | string | string[] | undefined;
    }
```

I don't think we should support `number`. Unsure why it supports `undefined`.

Reply:
Maybe we should define a `Header` type that is shared?

Reply:
That could probably be a good idea, unless we should just re-use OutgoingHttpHeaders?

Reply:
I think we should just define our own type. I don't like the implicit `number` conversion. The types in Node.js are made to fit how Node.js works, not to be great types. We can do better.

Reply:
It would be nice if we can pass in a header as a number and it gets turned into a string. To discuss later?